### DISCUSSION TITLE
how to install legacy build of pdfjs-dist by npm in react project?

### DISCUSSION THREAD

**Discussion by sujoys12** [mozilla/pdf.js#18300] [Q&A]
Promise.withResovers is breaking in older browsers. So I want to use the legacy build. But I have not seen any guide to install that through npm?

**Comment by nicolo-ribaudo** [comment]
It's in the same package, inside `pdfja-dist/legacy`

**Comment by ak37165** [comment]
I am trying to use the legacy build via typescript like so:
```
import {
	GlobalWorkerOptions,
	OnProgressParameters,
	PDFDocumentProxy,
	getDocument,
} from 'pdfjs-dist/legacy';
```

I get "Could not find a declaration file for module 'pdfjs-dist/legacy'" Is some .d.ts file missing?
The above import works if I import from 'pdfjs-dist' without the legacy suffix.

**Comment by ritschwumm** [comment]
this seems to be working:

`import { getDocument } from "pdfjs-dist/legacy/build/pdf.mjs"; `

**Reply by ak37165** [reply]
Thanks for sharing. Unfortunately even the legacy build does not work on iPhones older than 3 years (and a lot of people have those since iPhones don't break that quickly). So we had to go with an even older version of PDfjs and then one ends up having to include it via CDN instead of an npm package.

**Reply by ashispavan** [reply]
Could you please share the older version number that you ended up using? I am facing the same issue and was looking for an older version that works on older iPhones.

**Reply by ak37165** [reply]
We used 2.9.359. 

That is not a sustainable strategy though - what we are _thinking_ is using the latest version and on browsers where it does not work, display the file as images instead of PDF. The server will generate the images, using something like `pdf2image`. There won't be much ability to zoom for such browsers.

**Reply by ashispavan** [reply]
Ah I see. This helps. Thank you!
